spec/immutable-data-structures.html

```diff
@@ -6,27 +6,573 @@ <h1>Immutable Data Structures</h1>
+        <li>has the following properties:</li>
+      </ul>
+      <emu-clause id="sec-record.fromentries">
+        <h1>Record.fromEntries ( _iterable_ )</h1>
+        <p>The *fromEntries* function takes one argument _iterable_, and performs the following steps:</p>
+        <emu-alg>
+          1. Perform ? RequireObjectCoercible(_iterable_).
+          1. Let _fields_ be a new empty List.
+          1. Let _adder_ be a new Abstract Closure with parameters (_key_, _value_) that captures _fields_ and performs the following steps when called:
+            1. If Type(_value_) is Object, throw a *TypeError* exception.
+            1. Let _field_ be { [[Key]]: _key_, [[Value]]: _value_ }.
+            1. Append _field_ to the end of list _fields_.
+          1. Perform ! AddEntriesFromIterable(`undefined`, _iterable_, _adder_).
+          1. Return a new Record value whose [[Fields]] is _fields_.
+        </emu-alg>
+        <emu-note>
+          <p>The parameter _iterable_ is expected to be an object that implements an @@iterator method that returns an iterator object that produces a two element array-like object whose first element is a value that will be used as a Map key and whose second element is the value to associate with that key.</p>
+        </emu-note>
+      </emu-clause>
+      <emu-clause id="sec-record.isrecord">
+        <h1>Record.isRecord ( _arg_ )</h1>
+        <p>The *isRecord* function takes one argument _arg_, and performs the following steps:</p>
+        <emu-alg>
+          1. If Type(_arg_) is Record, return `true`.
+          1. If Type(_arg_) is Object and _arg_ has a [[RecordData]] internal slot, return `true`.
+          1. Return `false`.
+        </emu-alg>
+      </emu-clause>
+      <emu-clause id="sec-record.prototype">
+        <h1>Record.prototype</h1>
+        <p>The initial value of *Record.prototype* is the value *null*.</p>
+
+        <p>This property has the attributes { [[Writable]]: false, [[Enumerable]]: false, [[Configurable]]: false }.</p>
       </emu-clause>
     </emu-clause>
   </emu-clause>
 
   <emu-clause id="sec-tuple-objects">
     <h1>Tuple Objects</h1>
-    <p>TBD</p>
     <emu-clause id="sec-tuple-constructor">
       <h1>The Tuple Constructor</h1>
-      <p>TBD</p>
-      <emu-clause id="sec-tuple-constructor-tuple-value">
-        <h1>Tuple ( _value_ )</h1>
-        <p>TBD</p>
+      <p>The Tuple constructor:</p>
+      <ul>
+        <li>is the intrinsic object <dfn>%Tuple%</dfn>.</li>
+        <li>is the initial value of the *"Tuple"* property of the global object.</li>
+        <li>creates and initializes a new Tuple object when called as a function.</li>
+        <li>is not intended to be used with the `new` operator or to be subclassed. It may be used as the value of an *extends* clause of a class definition but a *super* call to the Tuple constructor will cause an exception.</li>
+      </ul>
+      <emu-clause id="sec-tuple-items">
+        <h1>Tuple ( ..._items_ )</h1>

```

Comment:
Hmm, do we want this to work, or do we want to require that people use Tuple.of for this functionality? They're sort of duplicated, and this isn't parallel to Array.

Reply:
For context, we had an out of band discussion about this and figured that it would make sense to, at least for now, do the same thing as `Array` here, where we create a `Tuple` full of `n` `undefined` values when called as `Tuple(n)`, the same single-argument-overload as `Array`.